tests/suites/hup/tests/storagemigration.js

```diff
@@ -0,0 +1,256 @@
+
+        test.is(
+          await this.context.get().worker.executeCommandInHostOS(
+            `balena ps --format '{{.ID}}' 2>/dev/null`,
+            this.context.get().link),
+          containerIDs,
+          "balenaEngine should run the previously running of containers"
+        );
+
+        await archiveMigrationLogs(this, test);
+        const diskSize = await this.context.get().worker.executeCommandInHostOS(
+            `balena image ls --format '{{.Size}}' | awk 'BEGIN{t=0; k=1000;M=k*1000;G=M*1000;} /kB$/{sub("kB","");t+=($0*k);} /MB$/{sub("MB","");t+=($0*M);} /GB$/{sub("GB","");t+=($0*G);} END{print t}'`,
+            this.context.get().link);
+        test.comment(`Total image size on disk: ${diskSize}`);
+        test.comment("Migration successful");
+
+      }
+    },
+    {
+      title: 'Successful migration and rollback-health triggered',
+      run: async function(test) {
+        await this.context.get().hup.initDUT(
+          this, test, this.context.get().link);
+
+        if (!(await shouldMigrate(this, test))) {
+          return; // SKIP
+        }
+
+        test.comment(`Creating data to migrate`);
+        await this.context.get().worker.executeCommandInHostOS(
+            `balena pull balenalib/${this.suite.deviceType.slug}-debian:buster && balena run -d --name test balenalib/${this.suite.deviceType.slug}-debian:buster balena-idle`,
+            this.context.get().link,
+          );
+
+        await this.context.get().hup.doHUP(this, test, 'image', this.context.get().hup.payload, this.context.get().link);
+
+        // break the inactive partition (new os) by replacing the openvpn binary
+        await this.context.get().worker.executeCommandInHostOS(
+            `cp /bin/bash $(find /mnt/sysroot/inactive/ | grep "bin/openvpn")`,
+            this.context.get().link,
+          );
+        // reduce number of failures needed to trigger rollback
+        await this.context.get().worker.executeCommandInHostOS(
+            `sed -i -e "s/COUNT=.*/COUNT=1/g" $(find /mnt/sysroot/inactive/ | grep "bin/rollback-health")`,
+            this.context.get().link,
+          );
+
+        await this.context.get().hup.doReboot(this, test, this.context.get().link);
+        await waitForRollbackHealth(this, test);
+
+        // NOTE how much does this overlap with the rollback tests we have in
+        // testlodge, implicitely I have to rely on that working as expected
+        // here and if it breaks it would also break this test...
+        test.is(
+          await this.context.get().worker.executeCommandInHostOS(
+            `[ -f /mnt/state/rollback-health-triggered ] || [ -f /mnt/state/rollback-health-failed ] && echo pass || echo fail`,
+            this.context.get().link),
+          "pass",
+          "Rollback-health should fail",
+        );
+
+        // sleep 10s
+        await new Promise(resolve => setTimeout(resolve, 11000));
+
+        // FIXME how do we wait for the rollback triggered reboot?
```

Comment:
Check the rollback-health service is `inactive` and the breadcrumbs are gone to avoid false-positives between reboots.

https://github.com/balena-os/meta-balena/blob/227a64eba0103bbc60324aa3782c6534373fadd1/tests/suites/hup/tests/rollback-health.js#L170-L187